### PR TITLE
Improve fmin/fmax

### PR DESCRIPTION
**Pull request by 2011eric**
Passed 2 more tests in riscv-arch-test.
Support of fmin and fmax is complete.

### CODE REVIEW

File: riscv_private.h [hunk 1]
```diff
@@ -333,6 +333,17 @@ static inline uint32_t calc_fclass(uint32_t f) {
 
   return out;
 }
+static inline bool is_nan(uint32_t f){
+  const uint32_t expn = f & FMASK_EXPN;
+  const uint32_t frac = f & FMASK_FRAC;
+  return (expn == FMASK_EXPN && frac);
+}
+
+static inline bool is_signan(uint32_t f){
```
Comment (jserv) [inline riscv_private.h]: In [Floating-point exceptions](https://www.ibm.com/docs/en/aix/7.3?topic=concepts-floating-point-exceptions) of IBM AIX Manual, signaling NaN is abbreviated as `snan`, which might represent better than "signan." In Linux, the prefix "sig" is used for UNIX signal specific identifiers. However, signaling NaN is quite different from them. Let's name it `is_snan` here.

File: riscv_private.h [hunk 2]
```diff
@@ -333,6 +333,17 @@ static inline uint32_t calc_fclass(uint32_t f) {
 
   return out;
 }
+static inline bool is_nan(uint32_t f){
```
Comment (jserv) [inline riscv_private.h]: Run `clang-format-12 -i *.[ch]` to indent.

**Review by jserv — CHANGES_REQUESTED** [review]
Rebase the latest "master" branch which checks C/C++ source files for consistent style.

**Review by jserv — CHANGES_REQUESTED** [review]
You should explain the IEEE754-201x semantics of fmin/fmax in git commit messages.

**Comment by 2011eric** [comment]
I think I can pack the code into a function, since the code of fmin and the one of fmax is similar.

**Comment by jserv** [comment]
> I think I can pack the code into a function, since the code of fmin and the one of fmax is similar.

You can introduce some macros as what [SoftFP](https://bellard.org/softfp/) does.

**Comment by jserv** [comment]
Fix the conflicts caused by recent directory structure changes.

File: src/riscv_private.h [hunk 3]
```diff
@@ -333,6 +333,17 @@ static inline uint32_t calc_fclass(uint32_t f) {
 
   return out;
 }
+static inline bool is_nan(uint32_t f){
```
Comment (jserv) [inline src/riscv_private.h]: Indent this file as well.

**Pull request merged**